mobile/test/java/org/chromium/net/CronetHttp3Test.java

```diff
@@ -0,0 +1,151 @@
+package org.chromium.net;
+
+import static org.chromium.net.testing.CronetTestRule.getContext;
+import static org.junit.Assert.assertEquals;
+
+import org.chromium.net.impl.CronvoyUrlRequestContext;
+import io.envoyproxy.envoymobile.engine.EnvoyEngine;
+import org.chromium.net.impl.CronvoyLogger;
+import androidx.test.core.app.ApplicationProvider;
+import androidx.test.filters.SmallTest;
+import org.chromium.net.impl.CronvoyUrlRequestContext;
+import org.chromium.net.impl.NativeCronvoyEngineBuilderImpl;
+import org.chromium.net.testing.CronetTestRule;
+import org.chromium.net.testing.CronetTestRule.CronetTestFramework;
+import org.chromium.net.testing.CronetTestRule.RequiresMinApi;
+import org.chromium.net.testing.Feature;
+import org.chromium.net.testing.TestUrlRequestCallback;
+import org.chromium.net.testing.TestUrlRequestCallback.ResponseStep;
+import io.envoyproxy.envoymobile.engine.JniLibrary;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+import io.envoyproxy.envoymobile.engine.testing.HttpTestServerFactory;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Collections;
+import java.io.File;
+import java.nio.file.Paths;
+import java.nio.file.Files;
+
+/**
+ * Test CronetEngine with production HTTP/3 logic
+ */
+@RunWith(RobolectricTestRunner.class)
+public class CronetHttp3Test {
+  @Rule public final CronetTestRule mTestRule = new CronetTestRule();
+
+  private static final String TAG = CronetHttp3Test.class.getSimpleName();
+
+  // URLs used for tests.
+
+  // If true, dump envoy logs on test completion.
+  // Ideally we could override this from the command line but that's TBD.
+  boolean mPrintEnvoyLogs = false;
+  // The HTTP/2 server, set up to alt-svc to the HTTP/3 server
+  private HttpTestServerFactory.HttpTestServer mHttp2TestServer;
```

Comment:
optional: In Java, the convention is to use camelCase and not Hungarian notation although there's a lot of existing code in Envoy Mobile that uses Hungarian notation. Since this is a new file, let's use camelCase.

Reply:
SG